### ISSUE TITLE
Kernel Panic on anything above 10.10

### ISSUE THREAD

**Issue by nunyabinez** [dokterdok/Continuity-Activation-Tool#375]
Have you tried [these steps](https://github.com/dokterdok/Continuity-Activation-Tool/wiki#cant-open-the-tool)?

Is there already an open topic for this issue?

Please include the following information.

- OS Version: 10.11.X and 10.12.2
- Mac model: Mac Pro 5,1
- Dongle or replaced card?: Dongle
- Used dongle/card: IOGEAR GBu521

Your Issue: The patch works with 10.10.5. But if I try either the main release or the beta on 10.11 or 10.12 it installs properly, but then kernel panics. I see people saying that this works on 10.11 and 10.12. Is it related to my machine being a Mac Pro? I.e. are the people that are getting this to work using iMacs and MacBooks?

Also, is this project still going to be active? I would like to keep my Mac Pro on latest releases, I would hate to have to decide between staying on Yosemite versus having handoff.

P.S. thanks for everything you've been doing. I'm sure it sucks to have people complain about something you are doing for fee.

Steps done to try to fix the issue: Tried every combination of main and beta, 10.10, 10.11 and 10.12. Also tried clean installs versus upgrades (e.g. got it working on 10.10.5, updated to 10.11 and patched, then panicked)

**Comment by dagaloni** [comment]
Since I updated to Mac OS 10.12.3 I noticed kernel panics so I restored to a time machine backup with Mac OS 10.12.2 but unfortunately I still have kernel panics. 

I tried to uninstall and install it again (using beta 5) with no luck.

```
  Kernel Extensions in backtrace:
         com.apple.driver.IOBluetoothHIDDriver(5.0.2f4)[E16D58DF-EDAD-3399-9680-238CF71C934B]@0xffffff7fb074e000->0xffffff7fb0767fff
            dependency: com.apple.iokit.IOHIDFamily(2.0.0)[8D5ADCF8-7C6B-3B6D-9C9A-E2850B1A0AB6]@0xffffff7fb0072000
            dependency: com.apple.iokit.IOBluetoothFamily(5.0.0f18)[3AA9411B-1D19-318E-A777-19A4F0DC9D02]@0xffffff7fb065b000

```

**Comment by sparky672** [comment]
I also have a Mac Pro model 5,1 with an IOGear GBU521 that successfully activated CAT under Yosemite.

Unfortunately for me, I activated CAT under Sierra 10.12.3 and now I cannot boot to anything but a grey screen.  :(

**Comment by dagaloni** [comment]
@sparky672 Hopefully this helps: boot with command + R to startup in [Recovery Mode](https://support.apple.com/en-us/HT201314). Then install the OS (this won't delete any files, it just reinstalls the OS). That should work.

**Comment by sparky672** [comment]
>  install the OS (this won't delete any files, it just reinstalls the OS). That should work.

After just first installing Sierra yesterday, I really didn't want to have to do that again!

So here's how I fixed it...

(remove BT dongle)

1.  Boot into another partition or boot disk if you have one.  _(You'll need to be able to mount and edit the original boot drive.)_

2.  Hopefully you have backups of the two original Kext files.  Move them back to `/System/Library/Extensions/` replacing the two new ones put there by CAT.

3.  Now reboot.  System no longer hangs or kernel panics, but after a successful boot, all wireless hardware will not be found.  _(Thanks to some clever Apple engineers, my wireless Apple mouse is still working despite no Bluetooth.)_  

4.  Note:  SIP is still disabled.  Run CAT option 3, "Uninstall".  _(Even though you've already put the original Kext files in place allowing you to reboot, option 3 rebuilds the Kext caches and a few other important things that will allow your original wireless hardware to get recognized on the next boot.)_

5.  Optional: When step 4 is complete and you're prompted to reboot, hold down command-R and enter Recovery Mode, where you can re-enable SIP by entering `csrutil enable` at a Terminal prompt.

6.  Reboot - everything back to normal without needing to reinstall the OS!

-----

After this experience, I'm totally done with CAT.  

Meanwhile, I'm going to be looking for the Mac Pro 5,1 hardware solution.  I know I need an Apple BCM94360CD card, but I'm still fuzzy on exactly which adapter is needed, extension cables, etc.  I've seen a few kits out there for over a hundred bucks but I'm hoping to find something less expensive.  

Does anyone know if this one will work?

http://www.ebay.com/itm/Genuine-Apple-WiFi-802-11ac-Upgrade-Kit-Adapter-For-All-Mac-Pro-1-1-5-1-10-12-/252428497776?hash=item3ac5e93370:g:6OcAAOSwqfNXiLnJ#viTabs_0

**Comment by dagaloni** [comment]
"After just first installing Sierra yesterday, I really didn't want to have to do that again"

You do know that reinstalling via Recovery Mode is just like a software update and that your method way more complicated. Reinstalling the OS is the safest and easiest method.

**Comment by sparky672** [comment]
> You do know that reinstalling via Recovery Mode is just like a software update

Yeah, but I didn't feel like doing another 5 GB download just to fix two botched files.  _(and I knew with 100% certainty that none of my other settings, files, documents, apps, etc. would be touched.)_

> your method way more complicated

About 10-15 minutes.  Took longer to type up the procedure.  

Anyway, somebody else may find my suggestion handy.

**Comment by evolutioneh** [comment]
Just wanted to add my experience with the same issue. Kernel panic upon boot/wake. Issue with the same kext placed by CAT on 10.12.3. For whatever reason, the tool did not create a proper backup of the kexts. So even after installation was having 3-4 panics before able to get to the desktop. I had to use a TimeMachine backup to restore my whole system. BT400. Will be returning it and maybe looking at a hardware solution.. or not all all.
